radio/src/gui/common/stdlcd/radio_sdmanager.cpp

```diff
@@ -105,13 +105,14 @@ void onSdManagerMenu(const char * result)
     memcpy(reusableBuffer.sdmanager.originalName, line, sizeof(reusableBuffer.sdmanager.originalName));
     char * ext = getFileExtension(line, SD_SCREEN_FILE_LENGTH+1);
     if (ext) {
+      size_t len = strlen(ext);
```

Comment:
Ultimately we need to find the strlen of the whole `line` including ext, regardless of if there is an ext. or not.  It's just quicker if we know the location of ext. (like before with `unsigned(strlen(ext) + ext - line)`),  vs like we have to do in the other conditional case with strlen(line). Maybe I'm missing something... 

However I have a couple other thoughts on this whole part... 

A) I realized that it still won't work with file extensions > 5 chars because of the limit now set in sdcard:getFileExtension()  (the LEN_FILE_EXTENSION_MAX).  Of course we could work around that by making a new param for getFileExtension() which defines max. length to bother searching (with default of the macro value). But it would still be some arbitrary value, or perhaps simply use the whole SD_SCREEN_FILE_LENGTH (and then again later while actually editing the name).

B) I've actually found it would be useful on some occasions to edit file extensions. On "wider" LCDs (like X9D/etc) maybe we could let the user just edit the whole file name? I think SD_SCREEN_FILE_LENGTH is 64 chars in 2.2 (32 in 2.1) while most files names being used in the system (img/sound/script) are limited to 6-10 chars anyway.  With added benefit of cutting out a couple dozen lines of code... and in fact make most of this discussion completely obsolete!  :-)